### PR TITLE
cypressTests:modified

### PR DESCRIPTION
**Pull request by yash6102000**

### CODE REVIEW

File: app/components/tests/TestTableItem.tsx [hunk 1]
```diff
@@ -61,7 +61,9 @@ const TestTableItem = ({
         </div>
         <div className="test-name-navigation w-4/12 cursor-pointer truncate  text-base font-medium leading-6 text-primary  ">
           <NavLink to={`/tests/${id}`} tabIndex={0} key={index}>
-            <span id="test-name-navigation">{testName}</span>
+            <span className="test-name-navigation" id="test-name-navigation">
```
Comment (anuragP04) [inline app/components/tests/TestTableItem.tsx:64]: ID should be unique, this is not unique element

File: app/routes/sections.tsx [hunk 2]
```diff
@@ -69,6 +69,7 @@ export const action: ActionFunction = async ({ request }) => {
   const description = formData.get('description')
 
   if (typeof name !== 'string' || name.length === 0) {
+    console.log('asdjkhajsdhkashdjhaksd', ...formData)
```
Comment (anuragP04) [inline app/routes/sections.tsx:72]: remove this console.log

File: app/components/members/MemberListItem.tsx [hunk 3]
```diff
@@ -30,27 +30,24 @@ export default function MemberListItem({
...
         <div className=" col-span-2 overflow-ellipsis break-all">
-          <span className=" text-base leading-6 text-gray-700">
+          <span className="memberName text-base leading-6 text-gray-700">
             {user.firstName} {user.lastName}
           </span>
         </div>
-        <div className="col-span-3 overflow-ellipsis break-all">
-          <span className=" text-base leading-6 text-gray-700">
+        <div className="memberMail col-span-3 overflow-ellipsis break-all">
+          <span className="text-base leading-6  text-gray-700">
```
Comment (DhruvCopods) [inline app/components/members/MemberListItem.tsx:40]: remove extra space

File: app/components/results/ResultTab.tsx [hunk 4]
```diff
@@ -8,7 +8,11 @@ const ResultTab = ({
   testId: string
 }) => {
   return (
-    <div className="flex gap-8" id="Group_By_Tests-test-candidate-list-tab">
+    <div
+      className="flex gap-8"
+      id="Group_By_Tests-test-candidate-list-tab"
+      data-cy="Group_By_Tests-test-candidate-list-tab"
```
Comment (DhruvCopods) [inline app/components/results/ResultTab.tsx:14]: use kebab case

File: app/components/sections/SectionCard.tsx [hunk 5]
```diff
@@ -34,7 +34,10 @@ const SectionCard = ({
       id="section-card"
     >
       <div className="flex items-center justify-between">
-        <h2 id="sctionName" className="text-xl font-semibold text-gray-700">
+        <h2
+          id="sectionName"
```
Comment (DhruvCopods) [inline app/components/sections/SectionCard.tsx:38]: use kabab case

File: app/components/tests/AddTest.tsx [hunk 6]
```diff
@@ -116,7 +116,12 @@ const AddTestComponent = ({ sections }: { sections: Array<TestSection> }) => {
     <div className="flex h-full flex-col gap-6 overflow-hidden">
       {/* header */}
       <header className="flex items-center justify-between">
-        <h2 role={testsConstants.addTestbutton} tabIndex={0} title="Add Test" className="text-3xl font-bold text-black">
+        <h2
+          role={testsConstants.addTestbutton}
+          tabIndex={0}
+          title="Add Test"
```
Comment (DhruvCopods) [inline app/components/tests/AddTest.tsx:122]: use constant

File: app/components/sections/add-question/AddQuestionInSection.tsx [hunk 7]
```diff
@@ -182,7 +182,7 @@ const AddQuestionInSection = () => {
       <BreadCrumb data={breadCrumbArray} />
       <h1
         title={sectionDetails?.name}
-        className="text-3xl font-bold leading-9 text-gray-900"
+        className="addQuestionHeading text-3xl font-bold leading-9 text-gray-900"
```
Comment (DhruvCopods) [inline app/components/sections/add-question/AddQuestionInSection.tsx:185]: remove this unused classname

File: cypress/e2e/group-by-test/groupByTestTable.cy.ts [hunk 8]
```diff
@@ -1,38 +1,32 @@
...
-      .type(cypress.email)
-      .should('have.value', cypress.email)
-    cy.get('#password')
-      .clear()
-      .type(cypress.password)
-      .should('have.value', cypress.password)
-    cy.findByRole('button').click()
-    cy.get('a').find('#Group_By_Tests').should('have.text', cypress.results).click()
+    cy.get('a')
+      .find('#Group_By_Tests')
```
Comment (DhruvCopods) [inline cypress/e2e/group-by-test/groupByTestTable.cy.ts:25]: id should be kabab case

File: cypress/e2e/group-by-test/groupByTestTable.cy.ts [hunk 9]
```diff
@@ -44,27 +38,20 @@ describe('Test for GroupByTestTable, Result', () => {
...
-      .type(cypress.email)
-      .should('have.value', cypress.email)
-    cy.get('#password')
-      .clear()
-      .type(cypress.password)
-      .should('have.value', cypress.password)
-    cy.findByRole('button').click()
-    cy.get('a').find('#Group_By_Tests').should('have.text', cypress.results).click()
+    cy.get('a', { timeout: 6000 })
+      .find('#Group_By_Tests')
```
Comment (DhruvCopods) [inline cypress/e2e/group-by-test/groupByTestTable.cy.ts:50]: id should be kabab case

File: cypress/e2e/group-by-test/groupByTestTable.cy.ts [hunk 10]
```diff
@@ -83,19 +70,12 @@ describe('Test for GroupByTestTable, Result', () => {
...
```
Comment (DhruvCopods) [inline cypress/e2e/group-by-test/groupByTestTable.cy.ts:74]: id should be kabab case

File: cypress/e2e/group-by-test/groupByTestTable.cy.ts [hunk 11]
```diff
@@ -105,27 +85,22 @@ describe('Test for GroupByTestTable, Result', () => {
...
```
Comment (DhruvCopods) [inline cypress/e2e/group-by-test/groupByTestTable.cy.ts:101]: id should be kabab case

File: cypress/e2e/group-by-test/groupByTestTable.cy.ts [hunk 12]
```diff
@@ -143,27 +118,22 @@ describe('Test for GroupByTestTable, Result', () => {
...
```
Comment (DhruvCopods) [inline cypress/e2e/group-by-test/groupByTestTable.cy.ts:134]: id should be kebab case

File: cypress/e2e/members-test/members.cy.ts [hunk 13]
```diff
@@ -3,123 +3,110 @@
...
-      .clear()
-      .type('careers@copods.co')
-      .should('have.value', cypress.email)
-    cy.get('#password')
-      .clear()
-      .type('kQuiz@copods')
-      .should('have.value', cypress.password)
-    cy.findByRole('button').click()
+  it('Test for conforming ,new member is added in a list or not', () => {
     cy.get('a').find('#Members').should('have.text', cypress.members).click()
```
Comment (DhruvCopods) [inline cypress/e2e/members-test/members.cy.ts:24]: first letter of id should be small

File: cypress/e2e/members-test/members.cy.ts [hunk 14]
```diff
@@ -3,123 +3,110 @@
...
-    cy.get('#email')
-      .clear()
-      .type('careers@copods.co')
-      .should('have.value', cypress.email)
-    cy.get('#password')
-      .clear()
-      .type('kQuiz@copods')
-      .should('have.value', cypress.password)
-    cy.findByRole('button').click()
     cy.get('a').find('#Members').should('have.text', cypress.members).click()
```
Comment (DhruvCopods) [inline cypress/e2e/members-test/members.cy.ts:40]: the first letter of id should be small

File: cypress/e2e/group-by-test/CandidateListOfTestItem.cy.ts [hunk 15]
```diff
@@ -1,35 +1,29 @@
...
+      .type(Cypress.env('password'))
+      .should('have.focus')
+      .should('have.value', Cypress.env('password'))
+    cy.get('[data-cy="submit"]').click()
+    cy.location('pathname').should('include', '/dashboard')
+  })
 
+  it('Check if that test is coming on group by test page or not', () => {
     cy.get('a', { timeout: 6000 })
       .find('#Group_By_Tests', { timeout: 6000 })
```
Comment (DhruvCopods) [inline cypress/e2e/group-by-test/CandidateListOfTestItem.cy.ts:21]: use kebab case

File: cypress/e2e/group-by-test/CandidateListOfTestItem.cy.ts [hunk 16]
```diff
@@ -1,35 +1,29 @@
...
-      .type('careers@copods.co')
-      .should('have.value', 'careers@copods.co')
-    cy.get('#password', { timeout: 6000 })
-      .clear()
-      .type('kQuiz@copods')
-      .should('have.value', 'kQuiz@copods')
-    cy.findByRole('button').click()
-
     cy.get('a', { timeout: 6000 })
       .find('#Group_By_Tests', { timeout: 6000 })
```
Comment (DhruvCopods) [inline cypress/e2e/group-by-test/CandidateListOfTestItem.cy.ts:28]: use kebab case

File: cypress/e2e/group-by-test/CandidateListOfTestItem.cy.ts [hunk 17]
```diff
@@ -41,17 +35,6 @@ describe('Visiting group by test of results page', () => {
...
-      .type('careers@copods.co')
-      .should('have.value', 'careers@copods.co')
-    cy.get('#password', { timeout: 6000 })
-      .clear()
-      .type('kQuiz@copods')
-      .should('have.value', 'kQuiz@copods')
-    cy.findByRole('button', { timeout: 6000 }).click()
-
     cy.get('a')
       .find('#Group_By_Tests', { timeout: 6000 })
```
Comment (DhruvCopods) [inline cypress/e2e/group-by-test/CandidateListOfTestItem.cy.ts:39]: use kebab case

File: cypress/e2e/members-test/members.cy.ts [hunk 18]
```diff
@@ -3,123 +3,110 @@
...
+
+    cy.get('.memberMail')
+      .contains(cypress.memberEmail)
+      .parent()
+      .parent()
+      .within(() => {
+        cy.get('#deleteButton').click()
+      })
+    cy.get('#deleteDialog').should('be.visible')
+    cy.get('#cancelDeletePopUp').should('have.text', cypress.cancel).click()
```
Comment (DhruvCopods) [inline cypress/e2e/members-test/members.cy.ts:69]: id should be kabab case

File: cypress/e2e/members-test/members.cy.ts [hunk 19]
```diff
@@ -3,123 +3,110 @@
...
 
-      return false
+    cy.get('.memberMail')
+      .contains(cypress.memberEmail)
+      .parent()
+      .parent()
+      .within(() => {
+        cy.get('#deleteButton').click()
+      })
+    cy.get('#deleteDialog').should('be.visible')
```
Comment (DhruvCopods) [inline cypress/e2e/members-test/members.cy.ts:93]: use kebab case for id

File: cypress/e2e/section-details-test/SectionDetails.cy.ts [hunk 20]
```diff
@@ -1,126 +1,93 @@
...
+//       .should('have.focus')
+//       .should('have.value', Cypress.env('email'))
+//     cy.get('input[name="password"]')
+//       .clear()
+//       .type(Cypress.env('password'))
+//       .should('have.focus')
+//       .should('have.value', Cypress.env('password'))
+//     cy.get('[data-cy="submit"]').click()
+//     cy.location('pathname').should('include', '/dashboard')
+//   })
```
Comment (DhruvCopods) [inline cypress/e2e/section-details-test/SectionDetails.cy.ts:22]: remove commented code

File: cypress/e2e/section-tests/Section.cy.ts [hunk 21]
```diff
@@ -2,115 +2,107 @@ import { cypress, routeFiles } from '~/constants/common.constants'
...
       .should('have.value', cypress.password)
-    cy.findByRole('button').click()
+    cy.get('[data-cy="submit"]').click()
+    cy.location('pathname').should('include', '/dashboard')
+  })
+
+  it('cancel Add section', () => {
     cy.get('a')
-      .find('#Sections')
+      .find('#Sections', { timeout: 8000 })
```
Comment (DhruvCopods) [inline cypress/e2e/section-tests/Section.cy.ts:23]: first letter of id should be small

File: cypress/e2e/section-tests/Section.cy.ts [hunk 22]
```diff
@@ -2,115 +2,107 @@ import { cypress, routeFiles } from '~/constants/common.constants'
...
-      .type('careers@copods.co')
-      .should('have.value', cypress.email)
-    cy.get('#password')
-      .clear()
-      .type('kQuiz@copods')
-      .should('have.value', cypress.password)
-    cy.findByRole('button').click()
+  it('Check Active State of Section', () => {
+    cy.get('a')
+      .find('#Sections')
```
Comment (DhruvCopods) [inline cypress/e2e/section-tests/Section.cy.ts:37]: forst letter of id should be small

File: cypress/e2e/section-tests/Section.cy.ts [hunk 23]
```diff
@@ -2,115 +2,107 @@ import { cypress, routeFiles } from '~/constants/common.constants'
...
+      .click()
+    cy.location('pathname', { timeout: 6000 }).should('include', '/sections')
+    cy.location().then((loc) => {
+      cy.location('search').should('include', loc.search)
+    })
+  })
 
+  it('Test for valid error message while adding new section without Title', () => {
     cy.get('a')
       .find('#Sections')
```
Comment (DhruvCopods) [inline cypress/e2e/section-tests/Section.cy.ts:48]: first letter of id should be small

File: cypress/e2e/section-tests/Section.cy.ts [hunk 24]
```diff
@@ -2,115 +2,107 @@ import { cypress, routeFiles } from '~/constants/common.constants'
...
-    cy.get('#password')
-      .clear()
-      .type('kQuiz@copods')
-      .should('have.value', cypress.password)
-    cy.findByRole('button').click()
+    cy.get('.Toastify__toast').should('have.text', cypress.nameIsReq)
+  })
+  it('Test for valid error message while adding new section without Description', () => {
     cy.get('a')
       .find('#Sections')
```
Comment (DhruvCopods) [inline cypress/e2e/section-tests/Section.cy.ts:63]: first letter of id should be small

File: cypress/e2e/section-tests/Section.cy.ts [hunk 25]
```diff
@@ -2,115 +2,107 @@ import { cypress, routeFiles } from '~/constants/common.constants'
...
-    cy.get('#password')
-      .clear()
-      .type('kQuiz@copods')
-      .should('have.value', cypress.password)
-    cy.findByRole('button').click()
+    cy.get('.Toastify__toast').should('have.text', cypress.descIsReq)
+  })
+  it('Test for valid error message while adding new section with duplicate Title', () => {
     cy.get('a')
       .find('#Sections')
```
Comment (DhruvCopods) [inline cypress/e2e/section-tests/Section.cy.ts:82]: first letter of id should be small

File: cypress/e2e/section-tests/Section.cy.ts [hunk 26]
```diff
@@ -2,115 +2,107 @@ import { cypress, routeFiles } from '~/constants/common.constants'
...
-      .clear()
-      .type('careers@copods.co')
-      .should('have.value', cypress.email)
-    cy.get('#password')
-      .clear()
-      .type('kQuiz@copods')
-      .should('have.value', cypress.password)
-    cy.findByRole('button').click()
     cy.get('a')
       .find('#Sections')
```
Comment (DhruvCopods) [inline cypress/e2e/section-tests/Section.cy.ts:100]: first letter of id should be small

File: cypress/e2e/side-nav-tests/SideNavTests.cy.ts [hunk 27]
```diff
@@ -1,106 +1,37 @@
...
-      .clear()
-      .type('kQuiz@copods')
-      .should('have.value', cypress.password)
-    cy.findByRole('button').click()
+    // sections page
+    cy.get('a').find('#Sections').should('have.text', cypress.Sections).click()
+    cy.location('pathname').should('eq', '/sections')
 
+    // members page
     cy.get('a').find('#Members').should('have.text', cypress.members).click()
```
Comment (DhruvCopods) [inline cypress/e2e/side-nav-tests/SideNavTests.cy.ts:34]: first letter of id should be small

File: cypress/e2e/smoke.cy.ts [hunk 28]
```diff
@@ -31,72 +34,84 @@ describe('smoke tests', () => {
...
-      .find('#Sections')
-      .should('have.text', routeFiles.sections)
-      .click()
-    cy.location('pathname', { timeout: 60000 }).should('include', '/sections')
+      .type(Cypress.env('password'))
+      .should('have.focus')
+      .should('have.value', Cypress.env('password'))
+    cy.get('[data-cy="submit"]').click()
+    cy.location('pathname').should('include', '/dashboard')
+    cy.get('a').find('#Sections').should('have.text', cypress.Sections).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:51]: first letter of id should be small

File: cypress/e2e/smoke.cy.ts [hunk 29]
```diff
@@ -31,72 +34,84 @@ describe('smoke tests', () => {
...
-    cy.get('button#submitButton', { timeout: 6000 })
-      .should('have.text', commonConstants.addButton)
-      .click()
-    cy.get('#section-cards').children().first().click()
+      .type(Cypress.env('password'))
+      .should('have.focus')
+      .should('have.value', Cypress.env('password'))
+    cy.get('[data-cy="submit"]').click()
+    cy.location('pathname').should('include', '/dashboard')
+    cy.get('a').find('#Sections').should('have.text', cypress.Sections).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:75]: first letter of id should be small

File: cypress/e2e/smoke.cy.ts [hunk 30]
```diff
@@ -31,72 +34,84 @@ describe('smoke tests', () => {
...
+        if (
+          el[0].getElementsByClassName('sectionName')[0].innerHTML ===
+          cypress.section1
+        ) {
+          cy.get('.sectionName').contains(cypress.section1).click()
+        }
+      })
+    })
+    cy.get('.sectionName').contains(cypress.section1).click()
     cy.get('#addQuestion').should('have.text', cypress.addQuest).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:111]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 31]
```diff
@@ -113,45 +128,109 @@ describe('smoke tests', () => {
     cy.get('#questionEditor #quillEditor').within(() => {
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:128]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 32]
```diff
@@ -113,45 +128,109 @@ describe('smoke tests', () => {
...
       cy.get('.ql-editor').type(`What is your Test Question ?`)
     })
+    cy.get('input[placeholder="Write your option here"]')
+      .clear()
+      .type('Option of question')
+    cy.get('#saveAndExit').click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:134]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 33]
```diff
@@ -113,45 +128,109 @@ describe('smoke tests', () => {
...
+      .should('have.focus')
+      .should('have.value', Cypress.env('email'))
+    cy.get('input[name="password"]')
+      .clear()
+      .type(Cypress.env('password'))
+      .should('have.focus')
+      .should('have.value', Cypress.env('password'))
+    cy.get('[data-cy="submit"]').click()
+    cy.location('pathname').should('include', '/dashboard')
+    cy.get('a').find('#Sections').should('have.text', cypress.Sections).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:150]: first letter of id should be small

File: cypress/e2e/smoke.cy.ts [hunk 34]
```diff
@@ -113,45 +128,109 @@ describe('smoke tests', () => {
...
+        if (
+          el[0].getElementsByClassName('sectionName')[0].innerHTML ===
+          cypress.section2
+        ) {
+          cy.get('.sectionName').contains(cypress.section2)
+        }
+      })
+    })
+    cy.get('.sectionName').contains(cypress.section2).click()
+    cy.get('#addQuestion').should('have.text', cypress.addQuest).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:162]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 35]
```diff
@@ -113,45 +128,109 @@ describe('smoke tests', () => {
...
-      .click()
-    cy.location('pathname', { timeout: 60000 }).should('include', '/tests')
-    cy.get('#addTest', { timeout: 10000 }).click()
-    cy.location('pathname', { timeout: 100000 }).should(
+      .type(Cypress.env('password'))
+      .should('have.focus')
+      .should('have.value', Cypress.env('password'))
+    cy.get('[data-cy="submit"]').click()
+    cy.location('pathname').should('include', '/dashboard')
+    cy.get('a').find('#Tests').should('have.text', testsConstants.Tests).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:202]: first letter of id should be small

File: cypress/e2e/smoke.cy.ts [hunk 36]
```diff
@@ -113,45 +128,109 @@ describe('smoke tests', () => {
...
+    cy.get('#addTest', { timeout: 6000 }).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:203]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 37]
```diff
@@ -113,45 +128,109 @@ describe('smoke tests', () => {
...
+      .clear()
+      .type(cypress.deleteTest1)
     cy.get('#quillEditor').within(() => {
       cy.get('.ql-editor').type(`Test Description`)
     })
-    cy.get('button#nextButton').should('have.text', 'Next').click()
-    cy.get('#0').find('hr').should('have.class', 'bg-primary')
-    cy.get('#1').find('hr').should('have.class', 'bg-primary')
+
+    cy.get('button#nextButton').should('have.text', cypress.next).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:216]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 38]
```diff
@@ -170,19 +249,143 @@ describe('smoke tests', () => {
       })
     })
     cy.get('button#nextButton').should('have.text', cypress.next).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:251]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 39]
```diff
@@ -170,19 +249,143 @@ describe('smoke tests', () => {
...
+        } else if (
+          el[0].getElementsByClassName('stepsName')[0].innerHTML ===
+          cypress.step3
+        ) {
+          cy.get('.stepsName').contains(cypress.step3)
+        }
+      })
+    })
+
+    cy.get('button#submitButton').should('have.text', cypress.submit).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:273]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 40]
```diff
@@ -170,19 +249,143 @@ describe('smoke tests', () => {
...
+      .should('have.value', Cypress.env('email'))
+    cy.get('input[name="password"]')
+      .clear()
+      .type(Cypress.env('password'))
+      .should('have.focus')
+      .should('have.value', Cypress.env('password'))
+    cy.get('[data-cy="submit"]').click()
+    cy.location('pathname').should('include', '/dashboard')
+    cy.get('a').find('#Tests').should('have.text', testsConstants.Tests).click()
+    cy.get('#addTest', { timeout: 6000 }).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:290]: first letter of id should be small and use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 41]
```diff
@@ -170,19 +249,143 @@ describe('smoke tests', () => {
...
+    cy.location('pathname', { timeout: 60000 }).should(
+      'include',
+      '/tests/add-test'
+    )
 
+    cy.get('input[placeholder="Enter test name"]', { timeout: 6000 })
+      .clear()
+      .type(cypress.test1)
+    cy.get('#quillEditor').within(() => {
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:299]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 42]
```diff
@@ -170,19 +249,143 @@ describe('smoke tests', () => {
...
+      cy.get('.ql-editor').type(`Test Description`)
+    })
+
+    cy.get('button#nextButton').should('have.text', cypress.next).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:303]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 43]
```diff
@@ -170,19 +249,143 @@ describe('smoke tests', () => {
...
+          cy.get('.stepsName').contains(cypress.step2)
+        }
+      })
+    })
+    // user reached to step 2
+
+    cy.get('div#section', { timeout: 60000 }).each((el) => {
+      cy.wrap(el).within(() => {
+        if (el.find('.count')[0].innerText != '0') {
+          cy.get('input#noOfQu').should('have.disabled', true)
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:324]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 44]
```diff
@@ -170,19 +249,143 @@ describe('smoke tests', () => {
...
+          cy.get('input#time').should('have.disabled', true)
+          cy.get('button')
+            .should('have.text', commonConstants.addButton)
+            .click()
+          cy.get('button').should('have.text', cypress.Remove)
+
+          cy.get('input#noOfQu').clear().type('1')
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:331]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 45]
```diff
@@ -170,19 +249,143 @@ describe('smoke tests', () => {
...
+          cy.get('input#time').clear().type('1')
+          cy.get('input#noOfQu').should('have.value', '1')
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:333]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 46]
```diff
@@ -170,19 +249,143 @@ describe('smoke tests', () => {
...
+          cy.get('input#time').should('have.value', '1')
+        }
+      })
+    })
+    cy.get('button#nextButton').should('have.text', cypress.next).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:338]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 47]
```diff
@@ -170,19 +249,143 @@ describe('smoke tests', () => {
...
+        } else if (
+          el[0].getElementsByClassName('stepsName')[0].innerHTML ===
+          cypress.step3
+        ) {
+          cy.get('.stepsName').contains(cypress.step3)
+        }
+      })
+    })
+
+    cy.get('button#submitButton').should('have.text', cypress.submit).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:360]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 48]
```diff
@@ -170,19 +249,143 @@ describe('smoke tests', () => {
...
+    cy.get('#email')
+      .clear()
+      .type('careers@copods.co')
+      .should('have.value', cypress.email)
+    cy.get('#password')
+      .clear()
+      .type('kQuiz@copods')
+      .should('have.value', cypress.password)
+    cy.findByRole('button').click()
+    cy.get('a').find('#Members').should('have.text', cypress.members).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:373]: first letter of id should be small

File: cypress/e2e/smoke.cy.ts [hunk 49]
```diff
@@ -170,19 +249,143 @@ describe('smoke tests', () => {
...
+    cy.get('#addMember').should('have.text', cypress.addMember).click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:374]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 50]
```diff
@@ -170,19 +249,143 @@ describe('smoke tests', () => {
...
+    cy.get('#AddPopUpModel').should('be.visible')
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:375]: use kebab case

File: cypress/e2e/smoke.cy.ts [hunk 51]
```diff
@@ -170,19 +249,143 @@ describe('smoke tests', () => {
...
+    cy.get('input[name="lastName"]')
+      .clear()
+      .type(cypress.memberLastName)
+      .should('have.value', cypress.memberLastName)
+    cy.get('input[name="email"]')
+      .clear()
+      .type(cypress.memberEmail)
+      .should('have.value', cypress.memberEmail)
+    cy.get('select').select('Recruiter')
+    cy.get('#addbutton').click()
```
Comment (DhruvCopods) [inline cypress/e2e/smoke.cy.ts:389]: use kebab case

File: cypress/e2e/group-by-test/CandidateListOfTestItem.cy.ts [hunk 52]
```diff
@@ -41,17 +35,6 @@ describe('Visiting group by test of results page', () => {
       .click()
```
Comment (anuragP04) [inline cypress/e2e/group-by-test/CandidateListOfTestItem.cy.ts:35]: where is assertion in this test?

File: cypress/e2e/tests/add-test.cy.ts [hunk 53]
```diff
@@ -1,61 +1,60 @@
...
+    cy.get('[data-cy="submit"]').click()
+    cy.location('pathname').should('include', '/dashboard')
+  })
 
-    cy.get('a', { timeout: 6000 })
-      .find('#Tests', { timeout: 6000 })
-      .should('have.text', testsConstants.Tests)
+  it('Visiting Add Test Page', () => {
+    cy.get('a').find('#Tests').should('have.text', testsConstants.Tests).click()
+    cy.get('#addTest', { timeout: 6000 })
```
Comment (DhruvCopods) [inline cypress/e2e/tests/add-test.cy.ts:26]: use kebab case

File: cypress/e2e/tests/add-test.cy.ts [hunk 54]
```diff
@@ -1,61 +1,60 @@
...
-    cy.findByRole('button').click()
-
-    cy.get('a', { timeout: 6000 })
-      .find('#Tests')
-      .should('have.text', testsConstants.Tests)
-      .click()
-    cy.location('pathname', { timeout: 60000 }).should('include', '/tests')
-    cy.get('#addTest').click()
+    cy.get('a').find('#Tests').should('have.text', testsConstants.Tests).click()
+    cy.get('#addTest', { timeout: 6000 }).click()
```
Comment (DhruvCopods) [inline cypress/e2e/tests/add-test.cy.ts:33]: first letter of id should be small and use kebab case

File: cypress/e2e/tests/add-test.cy.ts [hunk 55]
```diff
@@ -64,134 +63,96 @@ describe('Creating tests', () => {
...
```
Comment (DhruvCopods) [inline cypress/e2e/tests/add-test.cy.ts:67]: first letter should be small and us ekebab case

File: cypress/e2e/tests/add-test.cy.ts [hunk 56]
```diff
@@ -64,134 +63,96 @@ describe('Creating tests', () => {
...
```
Comment (DhruvCopods) [inline cypress/e2e/tests/add-test.cy.ts:79]: first letter should be small and use kebab case

File: cypress/e2e/tests/add-test.cy.ts [hunk 57]
```diff
@@ -64,134 +63,96 @@ describe('Creating tests', () => {
...
 
   it('Verify on clicking back button on step 2 user navigate back to step 2', () => {
-    cy.visit('/sign-in')
-    cy.get('#email')
-      .clear()
-      .type('careers@copods.co')
-      .should('have.value', cypress.email)
-    cy.get('#password')
+    cy.get('a').find('#Tests').should('have.text', testsConstants.Tests).click()
+    cy.get('#addTest', { timeout: 6000 }).click()
```
Comment (DhruvCopods) [inline cypress/e2e/tests/add-test.cy.ts:111]: first letter should be small and use kebab case

File: cypress/e2e/tests/add-test.cy.ts [hunk 58]
```diff
@@ -64,134 +63,96 @@ describe('Creating tests', () => {
...
-    cy.findByRole('button').click()
-
-    cy.get('a', { timeout: 6000 })
-      .find('#Tests')
-      .should('have.text', testsConstants.Tests)
-      .click()
-    cy.location('pathname', { timeout: 60000 }).should('include', '/tests')
-    cy.get('#addTest').click()
+    cy.get('a').find('#Tests').should('have.text', testsConstants.Tests).click()
+    cy.get('#addTest', { timeout: 6000 }).click()
```
Comment (DhruvCopods) [inline cypress/e2e/tests/add-test.cy.ts:134]: first letter should be small and use kebab case

File: app/constants/common.constants.ts [hunk 59]
```diff
@@ -147,9 +148,20 @@ export const candidateExam = {
   startSection: 'Start Now',
   examTitle: 'Pre-Interview Assessment',
   info: 'Info',
-  noOfQuestions: "No. of Questions"
+  noOfQuestions: 'No. of Questions',
 }
 export const cypress = {
+  section1: `Aptitude - section1`,
```
[the remark this answers is not in the document]
Comment (yash6102000) [reply app/constants/common.constants.ts:154]: i think we have to cuz we are using this at multiple places

File: app/components/DeletePopUp.tsx [hunk 60]
```diff
@@ -56,7 +56,7 @@ export default function DeletePopUp({
             >
               <Dialog.Panel className="relative transform overflow-hidden rounded-lg bg-white text-left shadow-xl transition-all sm:my-8 sm:w-full sm:max-w-lg">
                 <div
-                  id="deleteDialog"
+                  id="delete-dialog"
```
Comment (anuragP04) [inline app/components/DeletePopUp.tsx:59]: 😃

**Review by anuragP04 — APPROVED** [review]

**Pull request merged**